text/0000-inline-const.md

```diff
@@ -0,0 +1,322 @@
+- Feature Name: `inline_const`
+- Start Date: 2020-04-30
+- RFC PR: [rust-lang/rfcs#2920](https://github.com/rust-lang/rfcs/pull/2920)
+- Rust Issue: TBD
+
+# Summary
+[summary]: #summary
+
+Adds a new syntactical element called an "inline `const`", written as
+`const { ... }`, which instructs the compiler to execute the contents of the
+block at compile-time. An inline `const` can be used as an expression or
+anywhere in a pattern where a named `const` would be allowed.
+
+```rust
+use std::net::Ipv6Addr;
+
+fn mock_ip(use_localhost: bool) -> &'static Ipv6Addr {
+    if use_localhost {
+        &Ipv6Addr::LOCALHOST
+    } else {
+        const { &Ipv6Addr::new(0x2001, 0xdb8, 0, 0, 0, 0, 0, 0) }
+    }
+}
+
```

Comment:
```suggestion

// Below is the same as
// fn main() {
//     const HALF: usize = u32::MAX / 2 + 1;
//     match *x {
//         0..HALF => println!("low"),
//         HALF..=u32::MAX => println!("high"),
//     }
// }

```
Add an example of how was it without inline const to let readers understand how it would look like without it. We could also make this block below the below block.